tests/integration/test_agent.py

```diff
@@ -74,10 +74,9 @@ def validate_final_state(final_state: State | None, test_name: str):
 )
 def test_write_simple_script(current_test_name: str) -> None:
     task = "Write a shell script 'hello.sh' that prints 'hello'. Do not ask me for confirmation at any point."
-    args = parse_arguments()
 
     # Create the agent
-    agent = Agent.get_cls(args.agent_cls)(llm=LLM(LLMConfig()))
+    agent = Agent.get_cls(os.getenv('DEFAULT_AGENT'))(llm=LLM(LLMConfig()))
```

Comment:
Why try to get the setting from os env in a test? To test that it defaults correctly?

Reply:
this `test_agent.py` test is only called from the `regenerate.sh` - it set the agent via `DEFAULT_AGENT`, which will be previously captured from `config.default_agent`, but since we get rid of the global var now, an easy way to get it would be getting it from env var now.

https://github.com/OpenDevin/OpenDevin/blob/db81b08aceb6f2e08dbf0176bd9efc75e0133ed1/tests/integration/regenerate.sh#L173-L183

Reply:
The logic is a bit weird right now, I'm thinking if there's better way

Reply:
But since the original script is also written this way, we may as well keep using env var now for consistency

https://github.com/OpenDevin/OpenDevin/blob/db81b08aceb6f2e08dbf0176bd9efc75e0133ed1/tests/integration/test_agent.py#L35

Reply:
Oki 😃